frontend/packages/ceph-storage-plugin/src/components/bucket-class/modals/edit-backingstore-modal.tsx

```diff
@@ -99,19 +227,29 @@ const BucketClassEditModal = withHandlePromise<
         <ModalBody>
           <p className="nb-bc-modal__text">
             {t(
-              'ceph-storage-plugin~BackingStore represents a storage target to be used as the underlying storage for the data in Multicloud Object Gateway buckets.',
+              'ceph-storage-plugin~{{storeType}} represents a storage target to be used as the underlying storage for the data in Multicloud Object Gateway buckets.',
+              {
+                storeType: isNamespaceType ? 'NamespaceStore' : 'BackingStore',
+              },
             )}
           </p>
-          <BackingStoreSelection
-            namespace={bucketClass.metadata.namespace}
-            tier1Policy={policyA}
-            tier2Policy={policyB}
-            selectedTierA={selectedA}
-            selectedTierB={selectedB}
-            setSelectedTierA={setSelectedA}
-            setSelectedTierB={setSelectedB}
-            hideCreateBackingStore
-          />
+          {!isNamespaceType && (
+            <BackingStoreSelection
+              namespace={bucketClass.metadata.namespace}
+              tier1Policy={policyA}
+              tier2Policy={policyB}
+              selectedTierA={state.tier1BackingStore}
+              selectedTierB={state.tier2BackingStore}
+              setSelectedTierA={(selectedA) =>
+                dispatch({ type: 'setBackingStoreTier1', value: selectedA })
+              }
+              setSelectedTierB={(selectedB) =>
+                dispatch({ type: 'setBackingStoreTier2', value: selectedB })
+              }
+              hideCreateBackingStore
+            />
+          )}
+          {isNamespaceType && getNamespaceStorePage()}
```

Comment:
you need conditional here
isNamespaceType ? ... : ...

Reply:
changed it. I am just used to lot of ifs than terinary conditions :)